### PR TITLE
Repo utils test

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a new job to test our repo utils, which is only run when there is a change in said utils. This is to make sure community contributions to those don't break anything there. As an example, it also add a couple of tests for the test fetcher.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Looks good! `filter_tests` is a bit weird, when reading it I don't have a good idea of what it does. The previous `filter_pipeline_tests` was very simple to understand, on the contrary

File: tests/repo_utils/test_tests_fetcher.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+from git import Repo
+
+
+git_repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+sys.path.append(os.path.join(git_repo_path, "utils"))
+
+transformers_path = os.path.join(git_repo_path, "src", "transformers")
+# Tests are run against this specific commit for reproducibility
+GIT_TEST_SHA = "07f6690206e39ed7a4d9dbc58824314f7089bb38"
```
[the remark this answers is not in the document]
Comment (sgugger) [reply tests/repo_utils/test_tests_fetcher.py:27]: Can do!

**Review by ydshieh — APPROVED** [review]
Looks good to me, thank you!

File: utils/tests_fetcher.py [hunk 2]
```diff
@@ -620,20 +627,29 @@ def infer_tests_to_run(output_file, diff_with_last_commit=False, filters=None, j
...
         return
     with open(output_file, "r", encoding="utf-8") as f:
         test_files = f.read().split(" ")
 
-    if len(test_files) == 0:
+    if len(test_files) == 0 or test_files == [""]:
         print("No tests to filter.")
         return
+
+    print(test_files)
```
Comment (ydshieh) [inline utils/tests_fetcher.py:648]: Sorry @sgugger I forgot to add a comment here. I feel this is also a debugging line and you intended to remove it. Just a nit and your call.

Comment (sgugger) [reply utils/tests_fetcher.py:648]: Yes, this can be removed.

**Pull request merged**